### PR TITLE
[MISC] Add experimental max_connections config

### PR DESCRIPTION
**Pull request by dragomirp**
Adds experimental support for force setting max_connections.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -1468,7 +1468,7 @@ def _restart(self, event: RunWithLock) -> None:
             return
 
         try:
-            for attempt in Retrying(wait=wait_fixed(3), stop_after_delay=stop_after_delay(300)):
+            for attempt in Retrying(wait=wait_fixed(3), stop=stop_after_delay(300)):
```
Comment (dragomirp) [inline src/charm.py:1471]: Typo™

File: src/charm.py [hunk 2]
```diff
@@ -1547,8 +1547,14 @@ def update_config(self, is_creating_backup: bool = False) -> bool:
             logger.warning("Early exit update_config: Cannot connect to Postgresql")
             return False
 
+        # Use config value if set, calculate otherwise
+        if self.config.experimental_max_connections:
+            max_connections = self.config.experimental_max_connections
+        else:
+            max_connections = max(4 * os.cpu_count(), 100)
+
```
Comment (dragomirp) [inline src/charm.py:1555]: Doing this only here would trigger a restart on deploy with `experimental_max_connections`.  I can also inject it in the config file, by changing the internal library, but I would rather keep it KISS.

File: config.yaml [hunk 3]
```diff
@@ -402,3 +402,7 @@ options:
       Allowed values are: from 0 to 2000000000.
     type: int
     default: 150000000
+  experimental_max_connections:
+    type: int
+    description: |
+      [EXPERIMENTAL] Force set max_connections.
```
Comment (dragomirp) [inline config.yaml:408]: Should we add some kind of validation here?

**Review by lucasgameiroborges — APPROVED** [review]
LGTM!

File: config.yaml (hunk 3 above)
Comment (lucasgameiroborges) [reply config.yaml:408]: should we maybe enforce an upper limit?

Comment (dragomirp) [reply config.yaml:408]: It's a footgun config.

**Pull request merged**